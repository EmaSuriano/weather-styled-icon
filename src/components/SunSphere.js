import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import defaultTheme from './defaultTheme';

const Spin = keyframes`
  100% {
    transform: rotate(360deg);
  }
`;

const SunSphere = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  width: 2.5em;
  height: 2.5em;
  background: ${({ theme }) => theme.backgroundColor};
  margin: ${({ onTheSide }) => (onTheSide ? '-2em 1em' : '-1.25em')};
  border-radius: 50%;
  box-shadow: 0 0 0 0.375em ${({ theme }) => theme.sunColor};
  animation: ${Spin} 12s infinite linear;
`;

SunSphere.propTypes = {
  onTheSide: PropTypes.bool,
  theme: PropTypes.shape({
    sunColor: PropTypes.string,
    backgroundColor: PropTypes.string,
  }),
};

SunSphere.defaultProps = {
  onTheSide: false,
  theme: defaultTheme,
};

export default SunSphere;
