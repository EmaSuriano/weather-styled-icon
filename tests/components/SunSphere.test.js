import React from 'react';
import SunSphere from '../../src/components/SunSphere';
import { shallow } from 'enzyme';
import 'jest-styled-components';

describe('<SunSphere />', () => {
  const theme = {
    sunColor: 'red',
    backgroundColor: 'white',
  };

  it('should render div with theme', () => {
    const wrapper = shallow(<SunSphere theme={theme} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with margin when onTheSide is true', () => {
    const wrapper = shallow(<SunSphere theme={theme} onTheSide />);
    expect(wrapper).toMatchSnapshot();
  });

  xit('Spin should return keyframes with theme', () => {
    // TOOD: Find a proper way to test Animations!
  });
});
