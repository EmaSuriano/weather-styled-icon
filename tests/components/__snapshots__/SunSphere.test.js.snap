// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SunSphere /> should render div with theme 1`] = `
.c0 {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 2.5em;
  height: 2.5em;
  background: white;
  margin: -1.25em;
  border-radius: 50%;
  box-shadow: 0 0 0 0.375em red;
  -webkit-animation: cZxgpV 12s infinite linear;
  animation: cZxgpV 12s infinite linear;
}

<div
  className="c0"
/>
`;

exports[`<SunSphere /> should render with margin when onTheSide is true 1`] = `
.c0 {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 2.5em;
  height: 2.5em;
  background: white;
  margin: -2em 1em;
  border-radius: 50%;
  box-shadow: 0 0 0 0.375em red;
  -webkit-animation: cZxgpV 12s infinite linear;
  animation: cZxgpV 12s infinite linear;
}

<div
  className="c0"
/>
`;
